edgelet/support-bundle/src/support_bundle old.rs

```diff
@@ -0,0 +1,684 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::env;
+use std::ffi::OsString;
+use std::fs::File;
+use std::io::{Cursor, Seek};
+use std::path::Path;
+use std::process::Command as ShellCommand;
+
+use chrono::{DateTime, Local, NaiveDateTime, Utc};
+use failure::Fail;
+use futures::{Future, Stream};
+use tokio::prelude::*;
+use zip::{write::FileOptions, CompressionMethod, ZipWriter};
+
+use edgelet_core::{LogOptions, LogTail, Module, ModuleRuntime};
+
+use crate::error::{Error, ErrorKind};
+use crate::logs::pull_logs;
+
+pub fn make_bundle<M>(
+    output_location: OutputLocation,
+    log_options: LogOptions,
+    include_ms_only: bool,
+    verbose: bool,
+    iothub_hostname: Option<String>,
+    runtime: M,
+) -> Box<dyn Future<Item = (Box<dyn Read + Send>, u64), Error = Error> + Send>
+where
+    M: 'static + ModuleRuntime + Clone + Send + Sync,
+{
+    match output_location {
+        OutputLocation::File(location) => {
+            let writer = future::result(
```

Comment:
Why not just use tokio file?

Reply:
this is the old file that's being refactored